dbt/adapters/athena/impl.py

```diff
@@ -43,28 +43,36 @@ def convert_datetime_type(cls, agate_table: agate.Table, col_idx: int) -> str:
         return "timestamp"
 
     @available
-    def generate_s3_data_path(
+    def generate_models_s3_write_path(
         self,
         env_name: str,
         domain_name: str,
         schema_name: str,
         table_name: str,
-        run_time: Optional[str] = None,
     ) -> str:
         conn = self.connections.get_thread_connection()
         client = conn.credentials
 
-        if run_time is None:
-            return (
-                f"{client.s3_data_dir}/{env_name}/domain_name={domain_name}/"
-                f"database_name={schema_name}/table_name={table_name}"
-            )
-        else:
-            return (
-                f"{client.s3_data_dir}/{env_name}/domain_name={domain_name}/"
-                f"database_name={schema_name}/table_name={table_name}/"
-                f"run_time={run_time}"
-            )
+        return (
+            f"{client.s3_data_dir}/{env_name}/models/domain_name={domain_name}/"
+            f"database_name={schema_name}/table_name={table_name}"
+        )
+
+    @available
+    def generate_seeds_s3_write_path(
+        self,
+        env_name: str,
+        domain_name: str,
+        schema_name: str,
+        table_name: str,
+    ) -> str:
+        conn = self.connections.get_thread_connection()
+        client = conn.credentials
+
+        return (
+            f"{client.s3_data_dir}/{env_name}/seeds/domain_name={domain_name}/"
```

Comment:
I may have missed something, but it looks like these functions are identical apart from `/seeds/` v `/models/`

Could it just be one function with seeds vs models as an argument?

Also, maybe different as this is a fork, but ideally this would be unit tested - looks like the sort of string constructor it's worth having a check on, to stop future typos.

Reply:
The tricky thing about testing this method is that the `AthenaConnectionManager` used to create the connection inherits from dbt's `SQLConnectionManager` and I'm not sure how to go about mocking dbt classes. It might be possible to use `moto`, because the `AthenaConnectionManager` uses PyAthena. Anyway, what I'm saying is this might take a long time to get right. It's worth considering that if we continue to use the Athena adapter going forward, we'd have to do a lot of work to get it up to standard. By that I mean, there are no unit tests for methods or functions, we'd have to rework documentation to match where we've made changes that are applicable only to the MoJ, and there are no proper releases for it either. Unit testing this method could form part of that work?

Reply:
I've updated the method to take a boolean parameter and tested in `dev` and it's working as expected.